dist/mushroom-strategy.js

```diff
@@ -1,1335 +1,1205 @@
+      ) {
+        states.push(state);
+      }
+    }
+
+    return states;
+  }
+
+  /**
+   * Get device entities from the entity registry, filtered by area and domain.
+   *
+   * The entity registry is a registry where Home-Assistant keeps track of all entities.
+   * A device is represented in Home Assistant via one or more entities.
+   *
+   * The result excludes hidden and disabled entities.
+   *
+   * @param {areaEntity} area Area entity.
+   * @param {string} domain The domain of the entity-id.
+   *
+   * @return {hassEntity[]} Array of device entities.
+   */
+  static #getDeviceEntities(area, domain) {
+    // Get the ID of the devices which are linked to the given area.
+    const areaDeviceIds = this.#devices.filter(device => {
+      return device.area_id === area.area_id;
+    }).map(device => {
+      return device.id;
+    });
+
+    // Return the entities of which all conditions below are met:
+    // 1. The entity is linked to a device which is linked to the given area,
+    //    or the entity itself is linked to the given area.
+    // 2. The entity's domain matches the given domain.
+    // 3. The entity is not hidden and is not disabled.
+    return this.#entities.filter(entity => {
+      return (
+          (areaDeviceIds.includes(entity.device_id) || entity.area_id === area.area_id)
+          && entity.entity_id.startsWith(`${domain}.`)
+          && entity.hidden_by == null && entity.disabled_by == null
+      );
+    });
+  }
+
+  /**
+   * Generate a dashboard.
+   *
+   * Called when opening a dashboard.
+   *
+   * @param {dashBoardInfo} info Dashboard strategy information object.
+   * @return {Promise<{views: Object[]}>}
+   */
+  static async generateDashboard(info) {
+    this.#hassStates      = info.hass.states;
+    const strategyOptions = info.config.strategy.options || {};
+
+    // Query the registries of Home Assistant.
+    [this.#entities, this.#devices, this.#areas] = await Promise.all([
+      info.hass.callWS({type: "config/entity_registry/list"}),
+      info.hass.callWS({type: "config/device_registry/list"}),
+      info.hass.callWS({type: "config/area_registry/list"}),
+    ]);
+
+    // Override the properties of the Home Assistant areas with custom values.
+    this.#areas = this.#areas.map(area => {
+      return {...area, ...strategyOptions.areas?.[area.area_id]};
+    });
+
+    // Create Person cards.
+    const personCards = [];
+
+    // Collect person entities.
+    for (const person of this.#entities.filter(entity => entity.entity_id.startsWith("person."))) {
+      personCards.push({
+        type: "custom:mushroom-person-card",
+        layout: "vertical",
+        primary_info: "none",
+        secondary_info: "none",
+        icon_type: "entity-picture",
+        entity: person.entity_id,
+      });
+    }
+
+    // Create Area cards.
+    const areaCards = [];
+
+    for (let area of this.#areas) {
+      if (area.hidden !== true) {
```

Comment:
For some reason, with  this configuration I am unable to edit the room cards
```YAML
strategy:
  type: custom:mushroom-strategy
  options:
    areas:
      - family_room:
          name: Family Room
          icon: mdi:television
          icon_color: green
          extra_cards:
            - type: custom:mushroom-chips-card
              chips:
                - type: entity
                  entity: sensor.abia_s_phone_battery_temperature
                  icon: mdi:thermometer
                  icon_color: pink
              alignment: center
      - kitchen:
          name: Kitchen
          icon: mdi:silverware-fork-knife
          icon_color: red
views: []
```

Reply:
I forget to mention I've introduced a breaking change.
Areas need to be configured without the dash since `strategy.options.areas` shouldn't be an array of objects anymore, but an object of objects:

```yaml
strategy:
  type: custom:mushroom-strategy
  options:
    areas:
      family_room:
        name: Family Room
        icon: mdi:television
        icon_color: green
        extra_cards:
          - type: custom:mushroom-chips-card
            chips:
              - type: entity
                entity: sensor.abia_s_phone_battery_temperature
                icon: mdi:thermometer
                icon_color: pink
            alignment: center
      kitchen:
        name: Kitchen
        icon: mdi:silverware-fork-knife
        icon_color: red
views: []
```

Reply:
I always say that anything is possible 😉

However, we're trying to establish a solid code-base for now.
Although I've implemented #11, It would be better to finish up this PR first and then resolve the issues one by one.

I like the idea.
My advice is to create a new issue (Feature request).

Reply:
Now thinking about it more, I think we should leave the area configuration as it is creating a breaking change and offers less features than the original. Our goal for this PR is to optimize the code base so let's try to keep it that way. What do you think?

Reply:
> as it is creating a breaking change

As an application/library/etc. evolves, it will introduce breaking changes eventually.
Certainly if it depends on user input like a custom configuration.
E.g. Home Assistant introduces breaking changes with every update.

Breaking changes are inevitable at some point.
Better making a breaking change at an early stage than later on when having more users.

IMHO, don't be too afraid of it.
Just make sure you inform the users about them and how to migrate.

It's a minor breaking change anyway:
```
areas = [
    0 => [
        'name' => 'Family Room',
        'icon' => 'mdi:television',
    ],
    1 => [
        'name' => 'Kitchen',
        'icon' => 'mdi:silverware-fork-knife',
    ],
]

// Versus

areas = [
    'living_room_id' => [
        'name' => 'Family Room',
        'icon' => 'mdi:television',
    ],
    'kitchen_id' => [
        'name' => 'Kitchen',
        'icon' => 'mdi:silverware-fork-knife',
    ],
]
```

Which one looks more logical to you?
The second version is more conventional for dealing with entities which carry an id.
All the user has to do once, is to remove a - in front of each entry.

I predict at some point you require the same change for `entity_config` also.

> offers less features than the original. 

Au contraire. It gives the opportunity to offer more functionality.
E.g. hiding or manipulating an area.

Like this, you can access, add, remove, alter property values in a convenient way.

Under normal circumstances, I agree with implementing fixes, changes, etc. separately from each other.
I would have waited implementing #11, but the implementation of it, was a great optimization on its own which fitted in this PR.

Reply:
But it is already committed.
Reverting the current changes, will be difficult and a lot of hassle.

What about creating a new issue about other order of entities?
This way we can refer to this issue once it's implemented.

It is possible to implement it in this PR, but I recommend to finish this PR without it.

Just create a new issue for each feature/fix you want to implement.
When this PR is finished, we create a new PR for any issue which is needed to be resolved before the next release.
When satisfied, release a new version.